packages/shared/src/HeaderButton.js

```diff
@@ -35,7 +33,7 @@ export default class HeaderRightButton extends React.PureComponent<
   };
 
   render = () => {
-    const { onPress } = this.props;
+    const { onPress, left } = this.props;
```

Comment:
Instead of a prop I would have `HeaderButton.Left` and `HeaderButton.Right`. Something similar to:

```js
HeaderButton.Right = ({ style, ...rest }) => ({
  <HeaderButton {...rest} style={[props.style, styles.right]} />
})
```